verticapy/performance/vertica/collection/collection_tables.py

```diff
@@ -0,0 +1,584 @@
+    """
+
+    def __init__(self, table_name: str, table_schema: str, key: str) -> None:
+        self.name = table_name
+        self.schema = table_schema
+        self.key = key
+        self.import_prefix = "qprof_"
+        self.import_suffix = f"_{key}"
+        self.staging_suffix = "_staging"
+
+    def get_import_name_fq(self) -> str:
+        return self._get_import_name(fully_qualified=True)
+
+    def get_import_name(self) -> str:
+        return self._get_import_name(fully_qualified=False)
+
+    def get_super_proj_name_fq(self) -> str:
+        return f"{self._get_import_name(fully_qualified=True)}_super"
+
+    def _get_import_name(self, fully_qualified) -> str:
+        return (
+            f"{self.schema}.{self.import_prefix}{self.name}{self.import_suffix}"
+            if fully_qualified
+            else f"{self.import_prefix}{self.name}{self.import_suffix}"
+        )
+
+    def get_staging_name(self) -> str:
+        return f"{self.schema}.{self.import_prefix}{self.name}{self.staging_suffix}{self.import_suffix}"
+
+    def get_drop_table_sql(self) -> str:
+        # Drop table can be in the base class because it is
+        # DROP TABLE IF EXISTS {self.get_import_name_fq}
+        raise NotImplementedError(f"Drop table not implemented yet")
+
+    # Recall: abstract methods won't raise by default
+    @abstractmethod
+    def get_create_table_sql(self) -> str:
+        raise NotImplementedError(
+            f"get_create_table_sql is not implemented in the base class CollectionTable."
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def get_create_projection_sql(self) -> str:
+        raise NotImplementedError(
+            f"get_create_projection_sql is not implemented in the base class CollectionTable"
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def has_copy_staging(self) -> str:
+        raise NotImplementedError(
+            f"get_create_projection_sql is not implemented in the base class CollectionTable"
+            f" Current table name = {self.name} schema {self.schema}"
+        )
+
+    @abstractmethod
+    def copy_from_local_file(self, filename: str) -> str:
+        # This method will generate and run the copy statement
+        # It will copy to a staging table when necessary
+        raise NotImplementedError(f"")
+
+
+class AllTableNames(Enum):
+    COLLECTION_EVENTS = "collection_events"
+    COLLECTION_INFO = "collection_info"
+    DC_EXPLAIN_PLANS = "dc_explain_plans"
+    DC_QUERY_EXECUTIONS = "dc_query_executions"
+    DC_REQUESTS_ISSUED = "dc_requests_issued"
+    EXECUTION_ENGINE_PROFILES = "execution_engine_profiles"
+    EXPORT_EVENTS = "export_events"
+    HOST_RESOURCES = "host_resources"
+    QUERY_CONSUMPTION = "query_consumption"
+    QUERY_PLAN_PROFILES = "query_plan_profiles"
+    QUERY_PROFILES = "query_profiles"
+    RESOURCE_POOL_STATUS = "resource_pool_status"
```

Comment:
I wonder if it would be better to make the values integer numbers. Afaik, enum items can be directly compared, but it would be faster if we want to compare their values.

Reply:
Never mind, I noticed that you need the literal name of the tables.

Reply:
Indeed numbers are faster to compare, but I'm not sure we can observe the difference in the comparison cost.

If I make these all numbers, then I would need something that mapped the numbers to a literal table name somewhere... I decided to do it here

Reply:
Makes sense!